geemap/common.py

```diff
@@ -16167,3 +16167,160 @@ def is_on_aws():
+        map_type = map_type.lower().replace("google.", "").replace("google", "").strip()
+
+        if map_type not in allowed_map_types:
+            raise ValueError(
+                "mapType must be one of 'roadmap', 'satellite', 'terrain', 'hybrid', 'traffic', 'streetview'"
+            )
+
+    tile_args = {}
+
+    # Define the parameters for each map type
+    for m_type in allowed_map_types:
+
+        mapType = m_type
+        layerTypes = None
+
+        if m_type == "hybrid":
+            mapType = "satellite"
+            layerTypes = ["layerRoadmap"]
+        elif m_type == "terrain":
+            layerTypes = ["layerRoadmap"]
+        elif m_type == "traffic":
+            mapType = "roadmap"
+            layerTypes = ["layerTraffic"]
+        elif m_type == "streetview":
+            mapType = "roadmap"
+            layerTypes = ["layerStreetview"]
+
+        tile_args[m_type] = {
+            "mapType": mapType,
+            "language": language,
+            "region": region,
+            "layerTypes": layerTypes,
+            **kwargs,
+        }
+
+        if tile_args[m_type].get("layerTypes") is None:
+            del tile_args[m_type]["layerTypes"]
+
+    gmap_providers = {}
+
+    for m_type in allowed_map_types:
+
+        # If map_type is provided, only generate the specified map
+        if map_type is not None and m_type != map_type:
+            continue
+
+        args = tile_args[m_type]
+        response = requests.post(
+            f"https://tile.googleapis.com/v1/createSession?key={api_key}",
+            headers={"Content-Type": "application/json"},
+            json=args,
+        )
+
+        if response.status_code == 200:
+            res = response.json()
+            gmap_provider = TileProvider(
+                {
+                    "url": f"https://tile.googleapis.com/v1/2dtiles/{{z}}/{{x}}/{{y}}?session={res['session']}&key={{accessToken}}",
+                    "attribution": f"© Google {m_type.capitalize()}",
+                    "accessToken": api_key,
+                    "name": f"Google.{m_type.capitalize()}",
+                    "ext": res["imageFormat"],
+                    "tileSize": res["tileWidth"],
+                }
+            )
+            gmap_providers[m_type] = gmap_provider
+        else:
+            display(response.text)
+            gmap_provider = None
+
+    if map_type is None:  # Return all map types
+        return gmap_providers
+    else:  # Return the specified map type
+        return gmap_providers.get(map_type)
+
+
+def google_maps_api_key(token_name="MAPS_API_KEY"):
+    """
+    Retrieves the Google Maps API key from the environment or Colab user data.
+
+    Args:
+        token_name (str, optional): The name of the environment variable or Colab user data key where the API key is stored. Defaults to 'MAPS_API_KEY'.
+
+    Returns:
+        str: The API key, or None if it could not be found.
+    """
+    if in_colab_shell():
```

Comment:
It's possible for a user to set environment variables in a Colab shell. Let's do the following so we do a lookup in both places:

```
env_value = os.environ.get(token_name, None)
if in_colab_shell():
  return userdata.get(lookup_key, env_value)
return env_value
```

We shouldn't use a global to keep track of the Maps API key (`MAPS_API_KEY`).

Reply:
Unlike `os.environ`, Colab's `userdata.get` does not allow setting environment variable programmatically. So `userdata.get(lookup_key, env_value)` would not work here. Therefore, I only change the variable name from `MAPS_API_KEY` to `env_value`. 

![image](https://github.com/gee-community/geemap/assets/5016453/1ddffa35-01bf-4ae5-bbc8-73afbd5f7b7b)